From eec5a432b255671bbb3d7f8dd99ed1d1d9fcbe7e Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Thu, 11 Apr 2024 16:09:05 -0700
Subject: [PATCH] ==== young filter followup ====

Subject: [PATCH] mm/damon/paddr: do page level access check for pageout on its own

The 'pageout' DAMOS action implementation of 'paddr', a DAMON operations
set for the physical address space, asks reclaim_pages() to do page
level access check again.  But the user can ask 'paddr' to do the page
level access check on its own, using DAMOS filter of 'young page' type.
Meanwhile, 'paddr' is the only user of reclaim_pages() that ask the page
level access check.

Make 'paddr' does the page level access check on its own always, and
simplify reclaim_pages() to always skip the page level access check.  As
a result of the change for reclaim_pages(), reclaim_folio_list(), which
is called by reclaim_pages(), also no more need to do the page level
access check.  Simplify the function, too.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/tk1fLQP5 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/tk1fLQP5

diff --git a/damon_meta_changes/tk1fLQP5 b/damon_meta_changes/tk1fLQP5
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.2

